### PR TITLE
Spotify control component

### PR DESCRIPTION
**Pull request by nikolim**
Basic spotify control component to show the current track and play/pause button

### CODE REVIEW

**Review by Atema — COMMENTED** [review]
Looks nice and clean! Just one quick comment for now (will look at it more deeply later):

Instead of storing the images in the `public` folder, put them in the `Assets/images` folder, and then simply importing them into the file like follows:

```js
import playIcon from "Assets/images/play-icon.png";

<img src={playIcon} />
```

(You can take a look at the spinner in `WaitingView`, I did the same there.)

Then webpack will make sure everything is in the right place, optimise it for caching etc., and you can skip the hastle with `process.env.PUBLIC_URL`.

**Comment by nikolim** [comment]
Thanks, I was actually not sure how to do it properly :)

File: src/Components/SpotifyControl/SpotifyControl.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  const handleData = (playbackData) => {
+    if (playbackData.statusCode === 200) {
+      setData({
+        "image_src": playbackData.body.item.album.images[0].url,
+        "name": playbackData.body.item.name,
+        "artist": playbackData.body.item.artists.map((artist)=>artist.name).join(", "),
+        "is_playing": playbackData.body.is_playing,
+      });
+      setIsPlaying(playbackData.body.is_playing);
```
[the remark this answers is not in the document]
Comment (Atema) [reply src/Components/SpotifyControl/SpotifyControl.js:33]: Ánd there's also `playBackState`, which contains more of the same data.

[the remark this answers is not in the document]
Comment (Atema) [reply src/Components/SpotifyControl/SpotifyControl.js:33]: An option may be to just keep `playbackState` (pls get rid of the capital B :zipper_mouth_face:) for state management within the presenter, and than give the image, name, artist, and playing state as separate props to the view:

```jsx
<SpotifyControlView
  name={playbackState.item.name}
  artist={...
/>
```

File: src/Components/SpotifyControl/SpotifyControl.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  const [data, setData] = useState({
+    "image_src": process.env.PUBLIC_URL + "/bragi-icon.png",
+    "name": "Bragi",
+    "artist": "3000",
+    "is_playing": false,
+  });
+  const [isPlaying, setIsPlaying] = useState(false)
+
+  useEffect(() => {
+    if (token) {
```
Comment (Atema) [inline src/Components/SpotifyControl/SpotifyControl.js:20]: The `if (token)` check is technically not necessary, as long as the spotifyControl is only used on token-restricted pages.

File: src/Components/SpotifyControl/SpotifyControl.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      setPlayBackState(getPlaybackState(token.access_token).then(handleData));
```
Comment (Atema) [inline src/Components/SpotifyControl/SpotifyControl.js:21]: When doing the thing recommended below, this should be updated to something like:

```js
getPlaybackState(token.access_token)
  .then((newData) => {
    if (newData.statusCode === 200)
      setPlaybackState(newData); // Or newData.body, as the rest is not really necessary
  });

**Review by Atema — CHANGES_REQUESTED** [review]
As said, looks pretty good overall. The main thing left is that the same data is currently stored in several states, which is confusing as all have to be kept in sync (which it seems is not always done at the moment)

**Comment by Atema** [comment]
And one more thing: if the place you got the icons also has SVG versions, they'd be preferrable: they are usually smaller and scale better to different sizes.

**Review by Atema — APPROVED** [review]
LGTM

**Pull request merged**